frontend_tests/node_tests/composebox_typeahead.js

```diff
@@ -84,6 +84,8 @@ global.people.add({
     assert_typeahead_equals("hi emoji :", false);
     assert_typeahead_equals("hi emoji :ta", emoji_list);
     assert_typeahead_equals("hi emoji :da", emoji_list);
+    assert_typeahead_equals("hi emoji :da_", emoji_list);
+    assert_typeahead_equals("hi emoji :da ", emoji_list);
```

Comment:
```
I think it might be nice to add tests that more clearly show that :da_ and :da shouldn't match tada but possibly should match things like panda_face.
```
Yeah it will be good to add tests for these.